packages/core/src/analytics/integrations/Omnitracking/Omnitracking.js

```diff
@@ -215,26 +218,56 @@ class Omnitracking extends Integration {
         );
       }
       case analyticsTrackTypes.TRACK: {
-        precalculatedParameters = this.getPrecalculatedParametersForEvent(data);
+        const eventMapperFn = trackEventsMapper[data.event];
 
-        if (!this.validateTrackingRequisites()) {
-          logger.warn(
-            `[Omnitracking] - Event ${data.event} tried to track without an unique view id.
-                Consider tracking an event only after a page event.
-                This behaviour is not recommended and will not be supported in upcoming versions`,
+        if (eventMapperFn && !isFunction(eventMapperFn)) {
+          throw new Error(
+            '[Omnitracking] - The event mapper did not return a valid function. Please make sure the event is correctly mapped.',
           );
         }
 
-        return await this.postTracking(
-          formatTrackEvent(data, precalculatedParameters),
-          data,
-        );
+        if (eventMapperFn) {
+          const mappedEvents = eventMapperFn(data);
+
+          if (isArray(mappedEvents)) {
+            await Promise.all(
+              mappedEvents.map(async mappedEventData => {
+                await this.processTrackEvents(data, mappedEventData);
+              }),
+            );
+          }
+
+          if (isObject(mappedEvents)) {
+            return await this.processTrackEvents(data, mappedEvents);
```

Comment:
I think the await is unnecessary here because the return value is not being used by `analytics` when invoking the `track` function of all integrations.

Reply:
Fixed on dev-main-22-02-22. thanks 🙂